src/app/components/tree/tree.component.ts

```diff
@@ -1,13 +1,39 @@
-import {Component, Input} from '@angular/core';
-import {TreeItem} from "../../interfaces/TreeItem";
+import {Component, forwardRef, Input} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from "@angular/forms";
+import {Node} from "../../models/node";
 
 @Component({
   selector: 'app-tree',
   templateUrl: './tree.component.html',
-  styleUrls: ['./tree.component.scss']
+  styleUrls: ['./tree.component.scss'],
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => TreeComponent),
+    multi: true,
+  }],
 })
-export class TreeComponent {
+export class TreeComponent implements ControlValueAccessor {
+
+  @Input() tree: Node[] = [];
+
+  onChange2!: Function;
+  onTouch!: Function;
+
+  writeValue() {
+  }
+
+  registerOnChange(fn: Function): void {
+    this.onChange2 = fn;
+  }
+
+  registerOnTouched(fn: Function): void {
+    this.onTouch = fn;
+  }
+
+  select(item: Node): void {
+    item.operation(!item.isSelected)
+    this.onChange2(new Set<string>(Node.listSelectedItems))
```

Comment:
Why do you need to use Set here??? if you implement correct functions for changing this array, you will need it